frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/artifacts.cy.ts

```diff
@@ -152,8 +167,123 @@ describe('Artifacts', () => {
         .should('have.text', 'scalar metrics');
     });
   });
+
+  describe('artifact in pipeline run details page', () => {
+    it('only url text when both artifactsAPI and s3Endpoint are disabled', () => {
+      initPipelineTopologyIntercepts({ disableArtifactsAPI: true, disableS3Endpoint: true });
+      pipelineRunDetails.visit(
+        projectName,
+        mockPipeline.pipeline_id,
+        mockMetricsVisualizationVersion.pipeline_version_id,
+        mockMetricsVisualizationRun.run_id,
+      );
+
+      pipelineRunDetails.findTaskNode('markdown-visualization').click();
+
+      pipelineRunDetails
+        .findArtifactItems('markdown_artifact')
+        .should(
+          'contain.text',
+          's3://aballant-pipelines/metrics-visualization-pipeline/16dbff18-a3d5-4684-90ac-4e6198a9da0f/markdown-visualization/markdown_artifact',
+        );
+    });
+
+    it('url is clickable when artifact api is enabled', () => {
+      initPipelineTopologyIntercepts({ disableArtifactsAPI: false, disableS3Endpoint: false });
+      pipelineRunDetails.visit(
+        projectName,
+        mockPipeline.pipeline_id,
+        mockMetricsVisualizationVersion.pipeline_version_id,
+        mockMetricsVisualizationRun.run_id,
+      );
+
+      // Stub the window.open method to capture the URL
+      cy.window().then((win) => {
+        cy.stub(win, 'open').as('windowOpen');
+      });
+      pipelineRunDetails.findTaskNode('markdown-visualization').click();
+
+      pipelineRunDetails
+        .findArtifactItems('markdown_artifact')
+        .should(
+          'contain.text',
+          's3://aballant-pipelines/metrics-visualization-pipeline/16dbff18-a3d5-4684-90ac-4e6198a9da0f/markdown-visualization/markdown_artifact',
+        )
+        .click()
+        .then(() => {
+          cy.get('@windowOpen').should('be.called');
+          cy.get('@windowOpen')
+            .its('args.0.0') // To check args[0][0]
+            .then((arg0) => {
+              expect(arg0).to.equal(
+                'http://test-bucket.s3.dualstack.ap-south.amazonaws.com/metrics-visualization-pipeline',
+              );
+            });
+        });
+    });
+  });
```

Comment:
@christianvogt, I need your feedback on this test. To verify which URL has been called, I've used the window stubbing method since `cy.intercept` and `cy.wait` were not working. Please let me know if we need to change.

Reply:
I'm not familiar with testing a `window.open` call.
That being said, every link that is rendered on the page without some sort of dynamic behavior where the URL isn't known prior to rendering, should render a proper HTML anchor link. If that can be done then just test that the href attribute is correct without having to deal with actually navigating to the page.

Reply:
Got it, i've changed the functionality of calling window.open to anchor tag. Also updated the tests accordingly. Thanks